packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -175,17 +175,17 @@ class VMService {
   Future<Null> get done => _peer.done;
 
   // Events
-  Stream<ServiceEvent> get onDebugEvent => onEvent('Debug');
-  Stream<ServiceEvent> get onExtensionEvent => onEvent('Extension');
+  Future<Stream<ServiceEvent>> get onDebugEvent => onEvent('Debug');
+  Future<Stream<ServiceEvent>> get onExtensionEvent => onEvent('Extension');
   // IsolateStart, IsolateRunnable, IsolateExit, IsolateUpdate, ServiceExtensionAdded
-  Stream<ServiceEvent> get onIsolateEvent => onEvent('Isolate');
-  Stream<ServiceEvent> get onTimelineEvent => onEvent('Timeline');
+  Future<Stream<ServiceEvent>> get onIsolateEvent => onEvent('Isolate');
+  Future<Stream<ServiceEvent>> get onTimelineEvent => onEvent('Timeline');
   // TODO(johnmccutchan): Add FlutterView events.
 
   // Listen for a specific event name.
-  Stream<ServiceEvent> onEvent(String streamId) {
+  Future<Stream<ServiceEvent>> onEvent(String streamId) async {
     _streamListen(streamId);
-    return _getEventController(streamId).stream;
+    return new Future<Stream<ServiceEvent>>.value(_getEventController(streamId).stream);
```

Comment:
Shouldn't we `await` on the previous line and then just leave this line as it is?

Reply:
You are right.